tests/filterCategories.spec.mjs

```diff
@@ -0,0 +1,88 @@
+import { test, expect } from '@playwright/test';
+import mongoose from "mongoose";
+import dotenv from "dotenv";
+import categoryModel from '../models/categoryModel.js';
+import productModel from '../models/productModel.js';
+
+dotenv.config();
+
+test.beforeAll(async () => {
+    await mongoose.connect(process.env.MONGO_URL);
+    
+    const book = new productModel({
+        name: 'best-book',
+        slug: 'book-slug',
+        description: 'this is the best book',
+        price: 18,
+        category: new mongoose.Types.ObjectId('64b0c0f9a4b1a33d8e4a0d0e'),
+        quantity: 1,
+        photo: {
+            data: Buffer.from('sample photo data', 'utf-8'),
+            contentType: 'image/jpeg',
+        },
+        shipping: false,
+    });
+
+    const laptop = new productModel({
+        name: 'best-laptop',
+        slug: 'laptop-slug',
+        description: 'this is the best laptop',
+        price: 1500,
+        category: new mongoose.Types.ObjectId('74b0c0f9a4b1a33d8e4a0d0e'),
+        quantity: 1,
+        photo: {
+            data: Buffer.from('sample photo data', 'utf-8'),
+            contentType: 'image/jpeg',
+        },
+        shipping: false,
+    });
+
+    const bookCategory = new categoryModel({
+        name: 'books-category',
+        slug: 'books-category-slug',
+        _id: '64b0c0f9a4b1a33d8e4a0d0e',
+    });
+
+    const electronicsCategory = new categoryModel({
+        name: 'electronics-category',
+        slug: 'electronics-category-slug',
+        _id: '74b0c0f9a4b1a33d8e4a0d0e',
+    });
+
+    await book.save();
+    await laptop.save();
+    await bookCategory.save();
+    await electronicsCategory.save();
+});
+
+test.afterAll(async () => {
+    await productModel.deleteOne({ slug: 'book-slug' });
+    await productModel.deleteOne({ slug: 'laptop-slug' });
+    await categoryModel.deleteOne({ slug: 'books-category-slug' });
+    await categoryModel.deleteOne({ slug: 'electronics-category-slug' });
+    await mongoose.disconnect();
+});
+
+test('filter by categories', async ({ page }) => {
+    await page.goto('http://localhost:3000/');
```

Comment:
Do you want to assert all products are shown by default before doing any filtering? I.e. The book and laptop should be displayed intially, and all checkboxes for categories should be unchecked.

Reply:
I am already checking for both categories selected at once 👍